[Site Isolation] Mixed content handling

ruthvikkonda opened the pull request:
#### bd3009db79e1a9bc7b0bc40065872aae8653b485
<pre>
[Site Isolation] Mixed content handling
<a href="https://bugs.webkit.org/show_bug.cgi?id=300018">https://bugs.webkit.org/show_bug.cgi?id=300018</a>
<a href="https://rdar.apple.com/156109147">rdar://156109147</a>

Reviewed by Sihui Liu.

The current implementation of mixed content relies on frames being LocalFrames.
This assumption is no longer valid with Site Isolation. This PR fixes all cases
where RemoteFrames are involved in mixed content.

A new FrameTreeSyncData member (FrameURLProtocol) is introduced in order to solve
mixed content checks with RemoteFrames. Since provisional frames aren&apos;t in the frame tree
and we need to access parent frames of provisional frames, provisionalParentFrame() is added.

Since logging is now on a frame basis and is only allowed for LocalFrames, extra machinery is
added in order to enable logging for RemoteFrame scenarios in mixed content. For these scenarios,
we send an IPC message to WebPageProxy. WebPageProxy then performs its own addConsoleMessage()
which has been modified to forward the message to the appropriate WebProcess with the corresponding
LocalFrame. Rather than send(), we do sendToProcessContainingFrame(). This not only preserves existing
functionality that uses addConsoleMessage() but also enables all future use cases for RemoteFrames
that need to perform logging.

A representative sample of tests is added to tests/site-isolation. In order to test SI behavior,
these tests mainly include mixed content cases that involve cross site iframes.
A few of these tests don&apos;t use cross site iframes but are included in order to ensure that the main mixed
content scenarios still work with SI on.

Tests: http/tests/site-isolation/mixedContent/about-blank-iframe-in-main-frame.html
       http/tests/site-isolation/mixedContent/anchor-cross-origin-from-https-UpgradeMixedContent.html
       http/tests/site-isolation/mixedContent/beacon-cross-origin-UpgradeMixedContent.https.html
       http/tests/site-isolation/mixedContent/data-url-iframe-in-main-frame.html
       http/tests/site-isolation/mixedContent/iframe-upgrade-UpgradeMixedContent.https.html
       http/tests/site-isolation/mixedContent/insecure-audio-video-in-main-frame.html
       http/tests/site-isolation/mixedContent/insecure-css-in-iframe.html
       http/tests/site-isolation/mixedContent/insecure-iframe-in-iframe.html
       http/tests/site-isolation/mixedContent/insecure-iframe-in-main-frame-UpgradeMixedContent.html
       http/tests/site-isolation/mixedContent/insecure-iframe-in-sandboxed-iframe-UpgradeMixedContent.html
       http/tests/site-isolation/mixedContent/insecure-image-in-iframe-UpgradeMixedContent.html
       http/tests/site-isolation/mixedContent/insecure-script-in-iframe.html
       http/tests/site-isolation/mixedContent/insecure-websocket-in-iframe-UpgradeMixedContent.html
       http/tests/site-isolation/mixedContent/redirect-http-to-https-iframe-in-main-frame.html
       http/tests/site-isolation/mixedContent/redirect-https-to-http-iframe-in-main-frame.html
* LayoutTests/TestExpectations:
* LayoutTests/http/tests/site-isolation/mixedContent/about-blank-iframe-in-main-frame-expected.txt: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/about-blank-iframe-in-main-frame.html: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/anchor-cross-origin-from-https-UpgradeMixedContent-expected.txt: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/anchor-cross-origin-from-https-UpgradeMixedContent.html: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/beacon-cross-origin-UpgradeMixedContent.https-expected.txt: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/beacon-cross-origin-UpgradeMixedContent.https.html: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/data-url-iframe-in-main-frame-expected.txt: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/data-url-iframe-in-main-frame.html: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/iframe-upgrade-UpgradeMixedContent.https-expected.txt: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/iframe-upgrade-UpgradeMixedContent.https.html: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/insecure-audio-video-in-main-frame-expected.txt: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/insecure-audio-video-in-main-frame.html: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/insecure-css-in-iframe-expected.txt: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/insecure-css-in-iframe.html: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/insecure-iframe-in-iframe-expected.txt: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/insecure-iframe-in-iframe.html: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/insecure-iframe-in-main-frame-UpgradeMixedContent-expected.txt: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/insecure-iframe-in-main-frame-UpgradeMixedContent.html: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/insecure-iframe-in-sandboxed-iframe-UpgradeMixedContent-expected.txt: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/insecure-iframe-in-sandboxed-iframe-UpgradeMixedContent.html: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/insecure-image-in-iframe-UpgradeMixedContent-expected.txt: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/insecure-image-in-iframe-UpgradeMixedContent.html: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/insecure-script-in-iframe-expected.txt: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/insecure-script-in-iframe.html: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/insecure-websocket-in-iframe-UpgradeMixedContent-expected.txt: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/insecure-websocket-in-iframe-UpgradeMixedContent.html: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/redirect-http-to-https-iframe-in-main-frame-expected.txt: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/redirect-http-to-https-iframe-in-main-frame.html: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/redirect-https-to-http-iframe-in-main-frame-expected.txt: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/redirect-https-to-http-iframe-in-main-frame.html: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/resources/frame-with-about-blank-frame.html: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/resources/frame-with-insecure-css.html: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/resources/frame-with-insecure-frame.html: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/resources/frame-with-insecure-iframe.html: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/resources/frame-with-redirect-http-to-https-frame.html: Added.
* LayoutTests/platform/ios/TestExpectations:
* Source/WebCore/SaferCPPExpectations/UncheckedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebCore/dom/Document.cpp:
(WebCore::Document::populateDocumentSyncDataForNewlyConstructedDocument):
* Source/WebCore/loader/DocumentLoader.cpp:
(WebCore::DocumentLoader::willSendRequest):
* Source/WebCore/loader/LocalFrameLoaderClient.cpp:
(WebCore::LocalFrameLoaderClient::provisionalParentFrame const):
* Source/WebCore/loader/LocalFrameLoaderClient.h:
* Source/WebCore/loader/MixedContentChecker.cpp:
(WebCore::isDocumentSecure):
(WebCore::isDataContextSecure):
(WebCore::isMixedContent):
(WebCore::MixedContentChecker::shouldUpgradeInsecureContent):
(WebCore::MixedContentChecker::shouldBlockRequest):
(WebCore::logConsoleWarning):
* Source/WebCore/loader/MixedContentChecker.h:
* Source/WebCore/page/Frame.h:
* Source/WebCore/page/LocalFrame.cpp:
(WebCore::LocalFrame::reportMixedContentViolation const):
(WebCore::LocalFrame::frameURLProtocol const):
* Source/WebCore/page/LocalFrame.h:
* Source/WebCore/page/Page.cpp:
(WebCore::Page::updateProcessSyncData):
* Source/WebCore/page/ProcessSyncData.in:
* Source/WebCore/page/RemoteFrame.cpp:
(WebCore::RemoteFrame::reportMixedContentViolation const):
(WebCore::RemoteFrame::frameURLProtocol const):
* Source/WebCore/page/RemoteFrame.h:
* Source/WebCore/page/RemoteFrameClient.h:
* Source/WebKit/UIProcess/WebFrameProxy.cpp:
(WebKit::WebFrameProxy::calculateFrameTreeSyncData const):
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::addConsoleMessage):
(WebKit::WebPageProxy::reportMixedContentViolation):
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.messages.in:
* Source/WebKit/WebProcess/WebCoreSupport/WebLocalFrameLoaderClient.cpp:
(WebKit::WebLocalFrameLoaderClient::provisionalParentFrame const):
* Source/WebKit/WebProcess/WebCoreSupport/WebLocalFrameLoaderClient.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebRemoteFrameClient.cpp:
(WebKit::WebRemoteFrameClient::reportMixedContentViolation):
* Source/WebKit/WebProcess/WebCoreSupport/WebRemoteFrameClient.h:

Canonical link: <a href="https://commits.webkit.org/301022@main">https://commits.webkit.org/301022@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b4f19e271bc7867f8f91d8e9e845b093f11dfc5d

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/124610 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/44282 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/35016 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/131449 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/76547 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/f8097564-fdc3-4309-afa3-d4591b4ab225) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/126487 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/44986 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/52853 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/94803 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/62934 "") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/f65f5e00-d0c5-4622-b4d7-3814167a6de9) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/127564 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/35869 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/111439 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/75375 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/0001b505-0de6-4d48-8868-1eea000c745a) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/34800 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/29594 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/74928 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/105625 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/29825 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/134117 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/51459 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/39281 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/103279 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/51870 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/107661 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/103057 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26251 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/48420 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/26683 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/48388 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/51333 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/57129 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/50730 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/54082 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/52421 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

szewai commented on Source/WebCore/loader/MixedContentChecker.cpp:
```diff
@@ -30,72 +30,63 @@
 }
 
-static void logConsoleWarning(const LocalFrame& frame, bool blocked, const URL& target, bool isUpgradingIPAddressAndLocalhostEnabled)
+static void logConsoleWarning(const Frame& frame, bool blocked, const URL& target, bool isUpgradingIPAddressAndLocalhostEnabled)
```
We might just remove this function and invoke the one line below at callers

szewai commented on Source/WebKit/UIProcess/WebPageProxy.cpp:
```diff
@@ -16696,6 +16697,24 @@ void WebPageProxy::documentURLForConsoleLog(WebCore::FrameIdentifier frameID, Co
     completionHandler({ });
 }
 
+void WebPageProxy::reportMixedContentViolation(FrameIdentifier frameID, bool blocked, const URL& target, bool isUpgradingIPAddressAndLocalhostEnabled)
```
It looks like `isUpgradingIPAddressAndLocalhostEnabled` is set from `document->settings().iPAddressAndLocalhostMixedContentUpgradeTestingEnabled()`, so we might use `protectedPreferences()->iPAddressAndLocalhostMixedContentUpgradeTestingEnabled()` and remove the boolean from message (UI process should rely on its own record if possible)

szewai commented on Source/WebKit/UIProcess/WebPageProxy.cpp:
```diff
@@ -16696,6 +16697,24 @@ void WebPageProxy::documentURLForConsoleLog(WebCore::FrameIdentifier frameID, Co
 
+void WebPageProxy::reportMixedContentViolation(FrameIdentifier frameID, bool blocked, const URL& target, bool isUpgradingIPAddressAndLocalhostEnabled)
+{
+    RefPtr frame = WebFrameProxy::webFrame(frameID);
```
Early return if frame is null

szewai reviewed (approved):
r=me if bots are happy

ruthvikkonda committed:
[Site Isolation] Mixed content handling
https://bugs.webkit.org/show_bug.cgi?id=300018
rdar://156109147

Reviewed by Sihui Liu.

The current implementation of mixed content relies on frames being LocalFrames.
This assumption is no longer valid with Site Isolation. This PR fixes all cases
where RemoteFrames are involved in mixed content.

A new FrameTreeSyncData member (FrameURLProtocol) is introduced in order to solve
mixed content checks with RemoteFrames. Since provisional frames aren't in the frame tree
and we need to access parent frames of provisional frames, provisionalParentFrame() is added.

Since logging is now on a frame basis and is only allowed for LocalFrames, extra machinery is
added in order to enable logging for RemoteFrame scenarios in mixed content. For these scenarios,
we send an IPC message to WebPageProxy. WebPageProxy then performs its own addConsoleMessage()
which has been modified to forward the message to the appropriate WebProcess with the corresponding
LocalFrame. Rather than send(), we do sendToProcessContainingFrame(). This not only preserves existing
functionality that uses addConsoleMessage() but also enables all future use cases for RemoteFrames
that need to perform logging.

A representative sample of tests is added to tests/site-isolation. In order to test SI behavior,
these tests mainly include mixed content cases that involve cross site iframes.
A few of these tests don't use cross site iframes but are included in order to ensure that the main mixed
content scenarios still work with SI on.

Tests: http/tests/site-isolation/mixedContent/about-blank-iframe-in-main-frame.html
       http/tests/site-isolation/mixedContent/anchor-cross-origin-from-https-UpgradeMixedContent.html
       http/tests/site-isolation/mixedContent/beacon-cross-origin-UpgradeMixedContent.https.html
       http/tests/site-isolation/mixedContent/data-url-iframe-in-main-frame.html
       http/tests/site-isolation/mixedContent/iframe-upgrade-UpgradeMixedContent.https.html
       http/tests/site-isolation/mixedContent/insecure-audio-video-in-main-frame.html
       http/tests/site-isolation/mixedContent/insecure-css-in-iframe.html
       http/tests/site-isolation/mixedContent/insecure-iframe-in-iframe.html
       http/tests/site-isolation/mixedContent/insecure-iframe-in-main-frame-UpgradeMixedContent.html
       http/tests/site-isolation/mixedContent/insecure-iframe-in-sandboxed-iframe-UpgradeMixedContent.html
       http/tests/site-isolation/mixedContent/insecure-image-in-iframe-UpgradeMixedContent.html
       http/tests/site-isolation/mixedContent/insecure-script-in-iframe.html
       http/tests/site-isolation/mixedContent/insecure-websocket-in-iframe-UpgradeMixedContent.html
       http/tests/site-isolation/mixedContent/redirect-http-to-https-iframe-in-main-frame.html
       http/tests/site-isolation/mixedContent/redirect-https-to-http-iframe-in-main-frame.html
* LayoutTests/TestExpectations:
* LayoutTests/http/tests/site-isolation/mixedContent/about-blank-iframe-in-main-frame-expected.txt: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/about-blank-iframe-in-main-frame.html: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/anchor-cross-origin-from-https-UpgradeMixedContent-expected.txt: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/anchor-cross-origin-from-https-UpgradeMixedContent.html: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/beacon-cross-origin-UpgradeMixedContent.https-expected.txt: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/beacon-cross-origin-UpgradeMixedContent.https.html: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/data-url-iframe-in-main-frame-expected.txt: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/data-url-iframe-in-main-frame.html: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/iframe-upgrade-UpgradeMixedContent.https-expected.txt: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/iframe-upgrade-UpgradeMixedContent.https.html: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/insecure-audio-video-in-main-frame-expected.txt: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/insecure-audio-video-in-main-frame.html: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/insecure-css-in-iframe-expected.txt: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/insecure-css-in-iframe.html: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/insecure-iframe-in-iframe-expected.txt: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/insecure-iframe-in-iframe.html: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/insecure-iframe-in-main-frame-UpgradeMixedContent-expected.txt: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/insecure-iframe-in-main-frame-UpgradeMixedContent.html: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/insecure-iframe-in-sandboxed-iframe-UpgradeMixedContent-expected.txt: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/insecure-iframe-in-sandboxed-iframe-UpgradeMixedContent.html: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/insecure-image-in-iframe-UpgradeMixedContent-expected.txt: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/insecure-image-in-iframe-UpgradeMixedContent.html: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/insecure-script-in-iframe-expected.txt: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/insecure-script-in-iframe.html: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/insecure-websocket-in-iframe-UpgradeMixedContent-expected.txt: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/insecure-websocket-in-iframe-UpgradeMixedContent.html: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/redirect-http-to-https-iframe-in-main-frame-expected.txt: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/redirect-http-to-https-iframe-in-main-frame.html: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/redirect-https-to-http-iframe-in-main-frame-expected.txt: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/redirect-https-to-http-iframe-in-main-frame.html: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/resources/frame-with-about-blank-frame.html: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/resources/frame-with-insecure-css.html: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/resources/frame-with-insecure-frame.html: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/resources/frame-with-insecure-iframe.html: Added.
* LayoutTests/http/tests/site-isolation/mixedContent/resources/frame-with-redirect-http-to-https-frame.html: Added.
* LayoutTests/platform/ios/TestExpectations:
* Source/WebCore/SaferCPPExpectations/UncheckedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebCore/dom/Document.cpp:
(WebCore::Document::populateDocumentSyncDataForNewlyConstructedDocument):
* Source/WebCore/loader/DocumentLoader.cpp:
(WebCore::DocumentLoader::willSendRequest):
* Source/WebCore/loader/LocalFrameLoaderClient.cpp:
(WebCore::LocalFrameLoaderClient::provisionalParentFrame const):
* Source/WebCore/loader/LocalFrameLoaderClient.h:
* Source/WebCore/loader/MixedContentChecker.cpp:
(WebCore::isDocumentSecure):
(WebCore::isDataContextSecure):
(WebCore::isMixedContent):
(WebCore::MixedContentChecker::shouldUpgradeInsecureContent):
(WebCore::MixedContentChecker::shouldBlockRequest):
(WebCore::logConsoleWarning):
* Source/WebCore/loader/MixedContentChecker.h:
* Source/WebCore/page/Frame.h:
* Source/WebCore/page/LocalFrame.cpp:
(WebCore::LocalFrame::reportMixedContentViolation const):
(WebCore::LocalFrame::frameURLProtocol const):
* Source/WebCore/page/LocalFrame.h:
* Source/WebCore/page/Page.cpp:
(WebCore::Page::updateProcessSyncData):
* Source/WebCore/page/ProcessSyncData.in:
* Source/WebCore/page/RemoteFrame.cpp:
(WebCore::RemoteFrame::reportMixedContentViolation const):
(WebCore::RemoteFrame::frameURLProtocol const):
* Source/WebCore/page/RemoteFrame.h:
* Source/WebCore/page/RemoteFrameClient.h:
* Source/WebKit/UIProcess/WebFrameProxy.cpp:
(WebKit::WebFrameProxy::calculateFrameTreeSyncData const):
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::addConsoleMessage):
(WebKit::WebPageProxy::reportMixedContentViolation):
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.messages.in:
* Source/WebKit/WebProcess/WebCoreSupport/WebLocalFrameLoaderClient.cpp:
(WebKit::WebLocalFrameLoaderClient::provisionalParentFrame const):
* Source/WebKit/WebProcess/WebCoreSupport/WebLocalFrameLoaderClient.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebRemoteFrameClient.cpp:
(WebKit::WebRemoteFrameClient::reportMixedContentViolation):
* Source/WebKit/WebProcess/WebCoreSupport/WebRemoteFrameClient.h:

Canonical link: https://commits.webkit.org/301022@main
